rdt-app/src/main/java/io/ona/rdt_app/util/RDTJsonFormUtils.java

```diff
@@ -148,26 +149,37 @@ public void launchForm(Activity activity, String formName) throws JSONException
         launchForm(activity, formName, null);
     }
 
-    public void launchForm(Activity activity, String formName, String entityId) throws JSONException {
+    public void launchForm(Activity activity, String formName, Patient patient) throws JSONException {
         try {
             JSONObject formJsonObject = getFormJsonObject(formName, activity);
             String rdtId = Constants.Form.RDT_TEST_FORM.equals(formName) ? UUID.randomUUID().toString().substring(0, 5) : "";
-            prePopulateFormFields(formJsonObject, entityId, rdtId);
+            prePopulateFormFields(formJsonObject, patient, rdtId);
             startJsonForm(formJsonObject, activity, REQUEST_CODE_GET_JSON);
         } catch (JsonFormMissingStepCountException e) {
             Log.e(TAG, e.getStackTrace().toString());
         }
     }
 
-    public void prePopulateFormFields(JSONObject jsonForm, String entityId, String rdtId) throws JSONException, JsonFormMissingStepCountException{
-        jsonForm.put(ENTITY_ID, entityId);
+    public void prePopulateFormFields(JSONObject jsonForm, Patient patient, String rdtId) throws JSONException, JsonFormMissingStepCountException{
+        jsonForm.put(ENTITY_ID, patient == null ? null : patient.getBaseEntityId());
         JSONArray fields = getMultiStepFormFields(jsonForm);
         for (int i = 0; i < fields.length(); i++) {
             JSONObject field = fields.getJSONObject(i);
             if (Constants.Form.LBL_RDT_ID.equals(field.getString(KEY))) {
                 field.put(VALUE, rdtId);
                 field.put("text", "ID: " + rdtId);
             }
+            // pre-populate patient fields
+            if (patient != null) {
+                if (Constants.Form.LBL_PATIENT_NAME.equals(field.getString(KEY))) {
+                    field.put(VALUE, rdtId);
+                    field.put("text", patient.getPatientName());
+                }
+                if (Constants.Form.LBL_PATIENT_GENDER_AND_ID.equals(field.getString(KEY))) {
```

Comment:
This should be an `if else`

Reply:
Don't think so, it sets a different label from the one above.

Reply:
I see, it's a minor optimization. Done.